src/foam/u2/crunch/wizardflow/CapabilityAdaptAgent.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.u2.crunch.wizardflow',
+  name: 'CapabilityAdaptAgent',
+  documentation: `
+    If rootCapability in context is a string, then replace it with the real
+    Capability object from capabilityDAO.
+  `,
+
+  implements: [ 'foam.core.ContextAgent' ],
+
+  imports: [
+    'rootCapability as unadaptedRootCapability',
+    'capabilityDAO'
+  ],
+  exports: [ 'rootCapability' ],
+
+  properties: [
+    {
+      name: 'rootCapability',
+      class: 'FObjectProperty',
+      of: 'foam.nanos.crunch.Capability'
+    }
+  ],
+
+  methods: [
+    async function execute() {
```

Comment:
do problems arise  with this being an async/await function?

Reply:
Not that I'm aware of. `async` only affects what you can do inside the function, and also means it always returns a promise.